snapshots/lgbt_rights/2024-06-03/equaldex_extract.py

```diff
@@ -0,0 +1,549 @@
+            status = response.status_code
+
+            if status != 200:
+                log.error(f"{country}: Status {status}")
+
+        # Create a dictionary with the output
+        response_dict = json.loads(content)  # type: ignore
+
+        # Get the country name from the response
+        try:
+            country_name = response_dict["regions"]["region"]["name"]
+
+        except Exception:
+            country_name = ""
+
+        # Get the list of issues available
+        try:
+            issue_list = list(response_dict["regions"]["region"]["issues"].keys())
+
+        except Exception:
+            # If no issue list is found, add this country to the "countries with no data" list
+            countries_no_data.append(country)
+            issue_list = []
+
+        indices_data = pd.DataFrame()
+        for variable in VARIABLES_INDICES:
+            try:
+                indices_data.loc[0, variable] = response_dict["regions"]["region"][variable]
+            except Exception:
+                indices_data.loc[0, variable] = None
+
+        # Only concatenate if we have at least one column not null
+        if not indices_data.isnull().all().all():
+            # Concatenate data from previous countries with current country
+            df_indices = pd.concat([df_indices, indices_data], ignore_index=True)
+
+        # For each issue on the list
+        for issue in issue_list:
+            # Get the current status
+            try:
+                current_data = pd.DataFrame(
+                    response_dict["regions"]["region"]["issues"][issue]["current_status"],
+                    index=[0],
+                )
+
+            except Exception:
+                log.warning(f"{country_name}: No current data for {issue}")
+                current_data = pd.DataFrame()
+
+            # Get the history of the issue in the country
+            try:
+                historical_data = pd.DataFrame(
+                    response_dict["regions"]["region"]["issues"][issue]["history"],
+                )
+
+            except Exception:
+                log.warning(f"{country_name}: No historical data for {issue}")
+                historical_data = pd.DataFrame()
+
+            # Add country name column to the dataframe
+            current_data["country"] = country_name
+            historical_data["country"] = country_name
+
+            # Add issue column to the dataframe
+            current_data["issue"] = issue
+            historical_data["issue"] = issue
+
+            # Concatenate data from previous countries with current country
+            df_historical = pd.concat([df_historical, historical_data], ignore_index=True)
+            df_current = pd.concat([df_current, current_data], ignore_index=True)
+
+    # Error message with a summary of countries with no data
+    if countries_no_data:
+        log.error(f"Data was not found for the following {len(countries_no_data)} countries: \n{countries_no_data}")
```

Comment:
I defined the countries only because the API works country by country. This is written in case there are more countries without data, which can imply a problem in the extraction (beyond the API itself). I changed this to info instead of error.